src/protocols/WebSocket.ts

```diff
@@ -209,15 +211,61 @@ export default class WebSocketProtocol extends BaseProtocolRealtime {
     });
   }
 
+  enableCookieSupport () {
+    if (typeof XMLHttpRequest === 'undefined') {
+      throw new Error('Support for cookie cannot be enabled outside of a browser');
+    }
+
+    super.enableCookieSupport();
+    this._httpProtocol = new HttpProtocol(
+      this.host, 
+      {
+        port: this.port,
+        ssl: this.ssl,
+      }
+    );
+    this._httpProtocol.enableCookieSupport();
+  }
+
   /**
    * Sends a payload to the connected server
    *
    * @param {Object} payload
    */
-  send (request: RequestPayload) {
-    if (this.client && this.client.readyState === this.client.OPEN) {
-      this.client.send(JSON.stringify(request));
+  send (request: RequestPayload, options: JSONObject = {}) {
+    if (!this.client || this.client.readyState !== this.client.OPEN) {
+      return;
+    }
+
+    if ( this.cookieSupport
+      && request.controller === 'auth'
+      && ( request.action === 'login'
+        || request.action === 'logout'
+        || request.action === 'refreshToken')
+    ) {
+      const verifiedRequest = this._httpProtocol.formatRequest(request, options);
+
+      if (verifiedRequest) {
+        if (this.client) {
+          this.client.close();
+        }
+        this.client = null;
+        this.clientDisconnected(); // Simulate a disconnection, this will enable offline queue and trigger realtime subscriptions backup
```

Comment:
Does it enable the queuing of requests even if those options were not activated on Kuzzle constructor?

Reply:
No, if the offline queue is not enabled on Kuzzle constructor, then after the disconnection none of the requests will be queued.

Is there a case, in a well designed application, where someone would make a requests to `auth:login`, `auth:logout`, `auth:refreshToken` and then continue to make request without waiting for them to be resolved ?

Do you think that we should enable the offline queue when cookieAuth is enabled ?

Reply:
The user does not expect his protocol to be disconnected and if he enabled the offline queue then he does not expect to loose any request.

IMHO during the cookie Auth process (ws disconnect, http request, ws connect), we should queue requests and replay them when the websocket connection is open again.

@scottinet Wdyt?

Reply:
I mean if he enabled offline queue in the kuzzle constructor everything is fine, nothing should be loose since after the disconnection the offline queue will queue the following requests and replay them after being reconnected, so this is fine for this part.

But I was wondering, isn't it an undefined behaviour to make requests after `auth:login` `auth:logout` or `auth:refreshToken` without waiting for the `auth` request to be resolved ?
example:
```js
kuzzle.auth.login(...); // login is not awaited
kuzzle.index.create('nyc-open-data'); // We cannot ensure that index.create is not sent before auth.login has been resolved
```
Am I right ?

Reply:
Yes, developers should await but I'm thinking about a request made in another part of the application since the login action will take few event loop turn.

For login I'm not sure if it is a problem  but for a refreshToken for example the developer expect to never lost his connection to Kuzzle since he is refreshing his token at fixed intervals.

The whole process involve deconnection of the websocket connection so if requests are not queued some data will be loss. IMHO we should activate the offline queue and play back the requests even if the SDK was not constructed with this option